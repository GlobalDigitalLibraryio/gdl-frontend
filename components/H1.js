// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import styled from 'react-emotion';
import { textAlign } from 'styled-system';
import media from '../style/media';

const H1 = styled.h1`
  margin: 0;
  font-weight: 500;
  font-size: 26px;
  line-height: 30px;
  ${media.tablet`
    font-size: 36px;
    line-height: 42px;
  `};
  ${textAlign};
`;

export default H1;
