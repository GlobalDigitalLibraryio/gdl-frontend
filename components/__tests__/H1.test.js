// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';
import H1 from '../H1';

expect.addSnapshotSerializer(createSerializer(emotion));

test('Renders', () => {
  const tree = shallow(<H1>Header 1</H1>);

  expect(toJson(tree)).toMatchSnapshot();
});
