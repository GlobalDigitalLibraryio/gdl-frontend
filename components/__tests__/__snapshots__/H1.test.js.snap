// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
.emotion-0 {
  margin: 0;
  font-weight: 500;
  font-size: 26px;
  line-height: 30px;
}

@media (min-width:768px) {
  .emotion-0 {
    font-size: 36px;
    line-height: 42px;
  }
}

<h1
  className="emotion-0 emotion-1"
>
  Header 1
</h1>
`;
